File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.js
// Last updated: 29/05/2025, 21:42:41
/**
 * @param {number[]} numbers
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(numbers, target) {
    let left = 0
    let right = numbers.length - 1
    let result = []
    while (left < right) {
        let sum = numbers[left] + numbers[right]
        if (sum < target) {
            left++
            continue
        }
        if (sum > target) {
            right--
            continue
        }
        if (left < right && sum === target) {
            result.push(left + 1, right + 1)
            break
        }
    }
    return result
};

File: 167-TwoSumIiInputArrayIsSorted/167-TwoSumIiInputArrayIsSorted.py
# Last updated: 29/05/2025, 21:42:38
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
            left = 0
            right = len(numbers) - 1
            while left < right:
                current_sum = numbers[left] + numbers[right]
                if current_sum < target:
                    left += 1
                elif current_sum > target:
                    right -= 1
                else:
                    return [left + 1, right + 1]
            return []